extensions/dicom-seg/src/components/SegmentationPanel.js

```diff
@@ -0,0 +1,473 @@
+import cornerstoneTools from 'cornerstone-tools';
+import cornerstone from 'cornerstone-core';
+import moment from 'moment';
+
+import { utils } from '@ohif/core';
+import { ScrollableArea, TableList, Icon } from '@ohif/ui';
+
+import {
+  BrushColorSelector,
+  BrushRadius,
+  SegmentationItem,
+  SegmentItem,
+  SegmentationSelect,
+} from './index';
+
+import './SegmentationPanel.css';
+import SegmentationSettings from './SegmentationSettings';
+
+const { studyMetadataManager } = utils;
+
+/* TODO: wrap get/set interactions with the cornerstoneTools
+ * store with context to make these kind of things less blurry. */
+const segmentationModule = cornerstoneTools.getModule('segmentation');
+const { configuration } = segmentationModule;
+const DEFAULT_BRUSH_RADIUS = configuration.radius || 10;
+
+const refreshViewport = () => {
+  cornerstone.getEnabledElements().forEach(enabledElement => {
+    cornerstone.updateImage(enabledElement.element);
+  });
+};
+
+/**
+ * SegmentationPanel component
+ *
+ * @param {Object} props
+ * @param {Array} props.studies
+ * @param {Array} props.viewports - viewportSpecificData
+ * @param {number} props.activeIndex - activeViewportIndex
+ * @returns component
+ */
+const SegmentationPanel = ({ studies, viewports, activeIndex }) => {
+  const [brushRadius, setBrushRadius] = useState(DEFAULT_BRUSH_RADIUS);
+
+  /* TODO: We shouldn't hardcode this color, in the future the SEG may set the colorLUT to whatever it wants. */
+  const [brushColor, setBrushColor] = useState('rgba(221, 85, 85, 1)');
+  const [selectedSegment, setSelectedSegment] = useState();
+  const [showSegSettings, setShowSegSettings] = useState(false);
+  const [selectedSegmentation, setSelectedSegmentation] = useState();
+
+  const viewport = viewports[activeIndex];
+  const firstImageId = _getFirstImageId(viewport);
+  const { studyInstanceUid, seriesInstanceUid } = viewport;
+
+  /* CornerstoneTools */
+  const [brushStackState, setBrushStackState] = useState(
+    segmentationModule.state.series[firstImageId]
+  );
+
+  useEffect(() => {
+    setBrushStackState(segmentationModule.state.series[firstImageId]);
+  }, [studies, viewports, activeIndex, firstImageId]);
+
+  useEffect(() => {
+    if (brushStackState) {
+      setSelectedSegmentation(brushStackState.activeLabelmapIndex);
+    }
+
+    const labelmapModifiedHandler = event => {
+      console.warn('labelmap modified', event);
+      setBrushStackState(segmentationModule.state.series[firstImageId]);
+    };
+
+    /*
+     * These are specific to each element;
+     * Need to iterate cornerstone-tools tracked enabled elements?
+     * Then only care about the one tied to active viewport?
+     */
+    cornerstoneTools.store.state.enabledElements.forEach(enabledElement =>
+      enabledElement.addEventListener(
+        'cornersontetoolslabelmapmodified',
+        labelmapModifiedHandler
+      )
+    );
+
+    document.addEventListener('side-panel-change', handleSidePanelChange);
```

Comment:
👀 Can you explain what we're using this for? Maybe add a comment to explain. I don't 100% follow.

Reply:
I have used this so if I toggle some of the panels (open/hide), when toggling back to show the settings, the settings/seg component needs to be set back to its default values since the component doesnt rerender if panel is hidden. Maybe some other approach in mind?

Reply:
Event removed, now its directly using props from viewer.